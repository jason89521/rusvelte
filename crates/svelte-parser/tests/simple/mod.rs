use svelte_parser::Parser;
#[test]
fn test() {
    insta::glob!("inputs/*.svelte", |path| {
        let source = std::fs::read_to_string(path).unwrap();
        let allocator = oxc_allocator::Allocator::default();
        let mut parser = Parser::new(&source, &allocator);
        let result = parser.parse();
        assert!(result.is_ok());
        let root = result.unwrap();
        insta::assert_json_snapshot!(root)
    });
}
