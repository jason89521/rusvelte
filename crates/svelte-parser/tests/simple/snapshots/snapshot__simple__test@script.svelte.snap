---
source: crates/svelte-parser/tests/simple/mod.rs
expression: root
input_file: crates/svelte-parser/tests/simple/inputs/script.svelte
---
{
  "type": "Root",
  "start": 36,
  "end": 34,
  "fragment": {
    "type": "Fragment",
    "nodes": [
      {
        "type": "Text",
        "start": 34,
        "end": 36,
        "raw": "\n\n",
        "data": "\n\n"
      }
    ]
  },
  "module": {
    "type": "Script",
    "start": 0,
    "end": 34,
    "context": "module",
    "content": {
      "type": "Program",
      "start": 25,
      "end": 25,
      "sourceType": {
        "language": "javascript",
        "moduleKind": "module",
        "variant": "standard"
      },
      "hashbang": null,
      "directives": [],
      "body": []
    },
    "attributes": [
      {
        "type": "Attribute",
        "start": 8,
        "end": 24,
        "name": "context",
        "value": [
          {
            "type": "Text",
            "start": 17,
            "end": 23,
            "raw": "module",
            "data": "module"
          }
        ]
      }
    ]
  },
  "instance": {
    "type": "Script",
    "start": 36,
    "end": 53,
    "context": "default",
    "content": {
      "type": "Program",
      "start": 44,
      "end": 44,
      "sourceType": {
        "language": "javascript",
        "moduleKind": "module",
        "variant": "standard"
      },
      "hashbang": null,
      "directives": [],
      "body": []
    },
    "attributes": []
  }
}
